tests/unit/lms/services/d2l_api/client_test.py

```diff
@@ -28,6 +28,37 @@ def test_course_group_sets(
         D2LGroupSetsSchema.assert_called_once_with(basic_client.request.return_value)
         assert group_sets == d2l_group_sets_schema.parse.return_value
 
+    def test_group_set_groups(
+        self, svc, basic_client, D2LGroupsSchema, d2l_groups_schema, groups
+    ):
+        d2l_groups_schema.parse.return_value = groups
+
+        response = svc.group_set_groups("COURSE_ID", "GROUP_SET")
+
+        basic_client.request.assert_called_once_with(
+            "GET", "/COURSE_ID/groupcategories/GROUP_SET/groups/"
+        )
+        D2LGroupsSchema.assert_called_once_with(basic_client.request.return_value)
+        assert response == [dict(values, group_set_id="GROUP_SET") for values in groups]
```

Comment:
And having put a note about a list comprehension as a way of modifying a list above, I think it's good here.

I think brevity is probably better than speed in the tests.

The alternative mentioned below is some canned values.